lmdb only comes with a Makefile which only works on Linux so we create our own CMakeLists.txt file to be platform independent

diff -rupN orig/libraries/liblmdb/CMakeLists.txt patched/libraries/liblmdb/CMakeLists.txt
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ libraries/liblmdb/CMakeLists.txt	2026-06-12 11:51:51.369241534 +0200
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.26)
+project(lmdb C)
+
+find_package(Threads REQUIRED)
+
+add_library(lmdb STATIC mdb.c midl.c)
+target_include_directories(lmdb PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+target_link_libraries(lmdb PUBLIC Threads::Threads)
+set_target_properties(lmdb PROPERTIES ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+
+include(GNUInstallDirs)
+
+install(TARGETS lmdb ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES lmdb.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
